### ISSUE TITLE
Improvement: Don't check whenever(mock)-calls on verifyNoMoreInteractions(mock)

### ISSUE THREAD

**Issue by larmic** [nhaarman/mockito-kotlin#296]
```kotlin
@Test
internal fun `some testing`() {
     whenever(mock.ping()).thenReturn(Pong())
     pingService.sendPing()

     // this verification is already defined by whenever(...) 
     verify(mock).ping()
     // but is is needed, because otherwise this will fail
     verifyNoMoreInteractions(mock)
}
```

It will be reduce code if ```verifyNoMoreInteractions``` will not check ```whenever```-calls or this could be configurable by an optional parameter.

**Comment by bohsen** [comment]
> // this verification is already defined by whenever(...) 
     verify(mock).ping()

`verify(...)` checks that `mock.ping()` is invoked. `Whenever(...)` verifies nothing, but just defines what should happen **if** `mock.ping()` is invoked. And that depends on the implementation of `pingService.sendPing()`.

**Comment by larmic** [comment]
Yes you are right. But a ```verifyNoMoreInteractionsAndVerifyWhenever``` could help.

My tests often look like the one described above. And (for me) it feels like duplicating code.

**Comment by nhaarman** [comment]
Would Mockito's [Strictness](https://static.javadoc.io/org.mockito/mockito-core/2.22.0/org/mockito/quality/Strictness.html) API help here?
